### PR TITLE
Allow cacheback mode for disk devices

### PR DESCRIPTION
**Pull request by cedbossneo**
Signed-off-by: Cedric Hauber <hauber.c@gmail.com>

**What this PR does / why we need it**:

Today Kubevirt only support writethough cache.
Writeback allow less secure but better io speed.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Added Writeback cache support
```

### CODE REVIEW

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 1]
```diff
@@ -983,6 +983,8 @@ const (
 	CacheNone DriverCache = "none"
 	// CacheWriteThrough - I/O from the guest is cached on the host but written through to the physical medium.
 	CacheWriteThrough DriverCache = "writethrough"
+	// CacheWriteBack - I/O from the guest is cached on the host.
+	CacheWriteBack DriverCache = "writeback"
```
Comment (davidvossel) [inline staging/src/kubevirt.io/client-go/api/v1/types.go]: If we add this, we'll need to make sure to document the potential data integrity trade off that is occurring with this cache mode.

File: pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go [hunk 2]
```diff
@@ -1606,7 +1606,7 @@ func validateDisks(field *k8sfield.Path, disks []v1.Disk) []metav1.StatusCause {
 		}
 
 		// Verify if cache mode is valid
-		if disk.Cache != "" && disk.Cache != v1.CacheNone && disk.Cache != v1.CacheWriteThrough {
+		if disk.Cache != "" && disk.Cache != v1.CacheNone && disk.Cache != v1.CacheWriteThrough && disk.Cache != v1.CacheWriteBack {
```
Comment (davidvossel) [inline pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go]: are there any limitations with "writeback" cache that we should be checking for in the validation webhook?  basically, is there a configuration where someone can set cache=writeback and have their VMI fail to start? if so we should catch that here.

**Comment by rmohr** [comment]
@vladikr any insights?

**Comment by vladikr** [comment]
I'm sorry, I've missed this...

@davidvossel  I think migration is possible with the recent QEMU. Let me double-check and update.

File: staging/src/kubevirt.io/client-go/api/v1/types.go (hunk 1 above)
Comment (vladikr) [reply staging/src/kubevirt.io/client-go/api/v1/types.go:987]: + There is a good summary in https://docs.openstack.org/ocata/config-reference/compute/config-options.html

**Comment by vladikr** [comment]
It seems (according to [1]) that migration with writeback will be allowed in libvirt 5.6, that we are not yet using.

[1] https://bugzilla.redhat.com/show_bug.cgi?id=1660575

**Comment by vladikr** [comment]
We should be able to migrate workloads with disks that have write-back cache, after updating libvirt and qemu.
@cedbossneo would you be able to implement the changes @davidvossel has recommended ( except for the second and the third bullet)?

**Comment by Szpadel** [comment]
any updates on that?

**Pull request closed without merging**